routes/attachments.js

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Microsoft Corporation and others. Licensed under the MIT license.
+// SPDX-License-Identifier: MIT
+
+const asyncMiddleware = require('../middleware/asyncMiddleware')
+const express = require('express')
+const router = express.Router()
+const ResultCoordinates = require('../lib/resultCoordinates')
+
+// Get a proposed patch for a specific revision of a component
+router.get('/:token', asyncMiddleware(getAttachment))
+
+async function getAttachment(request, response) {
+  // TODO this is a hack. The coordinates are not intended to be sparse. Here make sure the
+  // token is the "namespace" to avoid the '-' showing up. Fix is likely to have an explicit
+  // AttachmentCoordinates class that has the right shape and generalize the stores to just take
+  // "coordinates" and not worry about how they are structured.
+  const coordinates = new ResultCoordinates('attachment', null, request.params.token)
+  const result = await harvestStore.get(coordinates)
```

Comment:
we should probably pass a stream to the store as a 2nd parameter.  Otherwise all attachments will get parsed as JSON

azblob looks like this:

```
    if (stream)
      return new Promise((resolve, reject) => {
        this.blobService.getBlobToStream(this.containerName, name, stream, responseOrError(resolve, reject))
      })
    return new Promise((resolve, reject) => {
      this.blobService.getBlobToText(this.containerName, name, resultOrError(resolve, reject))
    }).then(
      result => JSON.parse(result),
      error => {
        if (error.statusCode === 404) return null
        throw error
      }
    )

```

Reply:
ohhhh i see - the attachment is a property of the JSON blob. Should we be worried about contentType at all? Or is this only for text attachments?

Reply:
Don't expect complex files here. simple text. md and html at best